cmd/lncli/commands.go

```diff
@@ -250,6 +250,40 @@ func connectPeer(ctx *cli.Context) error {
 	return nil
 }
 
+var disconnectCommand = cli.Command{
+	Name:      "disconnect",
+	Usage:     "disconnect a remote lnd peer by peerID or pubKey",
+	ArgsUsage: "0@<pubkey>",
+	Action:    disconnectPeer,
+}
+
+func disconnectPeer(ctx *cli.Context) error {
+	ctxb := context.Background()
+	client, cleanUp := getClient(ctx)
+	defer cleanUp()
+
+	targetAddress := ctx.Args().First()
+	splitAddr := strings.Split(targetAddress, "@")
+	if len(splitAddr) != 2 {
```

Comment:
With taking into account previous comment this logic should be simplified to parsing the `btcec.PublicKey`.